app/services/idv/data_url_image.rb

```diff
@@ -1,16 +1,34 @@
 module Idv
   class DataUrlImage
     def initialize(data_url)
-      @header, base64_data = data_url.split(',', 2)
-      @data = Base64.decode64(base64_data || '')
+      header, data = URI(data_url.chomp).opaque.to_s.split(',', 2)
+      @header = header.to_s
+      @data = data.to_s
     end
 
+    BASE64_CONTENT_TYPE = /;base64$/.freeze
+
+    # @return [String]
     def content_type
-      @header.gsub(/^data:/, '').gsub(/;base64$/, '')
+      content_type, *_rest = @header.split(';')
+      content_type.to_s
     end
 
+    # @return [String]
     def read
-      @data
+      if base64_encoded?
+        Base64.decode64(@data)
+      else
+        # rubocop:disable Lint/UriEscapeUnescape
+        URI.decode(@data)
+        # rubocop:enable Lint/UriEscapeUnescape
+      end
+    end
+
+    private
+
+    def base64_encoded?
+      !!@header.match(BASE64_CONTENT_TYPE)
```

Comment:
May I suggest the usage of `match?`.
This looks like a method required to return a boolean value, match will return an object or `nil`, the double negation would force a boolean, but `match?` would do the same work more straightforward.

Reply:
Good point! Made a new PR: https://github.com/18F/identity-idp/pull/4081